lib/battle/Unit.h

```diff
@@ -25,6 +25,16 @@ class JsonSerializeFormat;
 
 namespace battle
 {
+
+enum BattlePhases
+{
+	SIEGE, // [0] - turrets/catapult,
+	NORMAL, // [1] - normal (unmoved) creatures, other war machines,
+	WAIT_MORALE, // [2] - waited creatures that had morale,
+	WAIT, // [3] - rest of waited creatures
+	MAX_NO_OF_PHASES // [4] - number of phases.
+};
```

Comment:
Non-class enum declarations should be put into their own namespace to keep global namespace (or in this case - `battle` namespace) clean.
```suggestion
namespace BattlePhases
{
	enum Type
	{
		SIEGE, // [0] - turrets/catapult,
		NORMAL, // [1] - normal (unmoved) creatures, other war machines,
		WAIT_MORALE, // [2] - waited creatures that had morale,
		WAIT, // [3] - rest of waited creatures
		MAX_NO_OF_PHASES // [4] - number of phases.
	};
}
```

Reply:
Yes, enum class is better but this enum is used for std::array indexing & for iteration. Enum class would need too many casts with this usage scenario IMO.

> also I'd assign values directly instead of keeping them inside comments
I'd rather say that values are unnecessary here & can be removed entirely

This enum is not mapped to H3 indexes or anything like that, so enum values can be added/removed safely, even if this changes their numeric values.

Reply:
Now I understood what you wanted with namespace for enum. Thought having it in battle was enough.
Fixed as suggested and deleted enum values from comments.

Reply:
"Namespace enum" is something often used in VCMI code (was "invented" way before c++11) so kind of expected you to know about it.